pkg/controller/perconaservermongodb/external_service.go

```diff
@@ -34,6 +35,65 @@ func (r *ReconcilePerconaServerMongoDB) ensureExternalServices(cr *api.PerconaSe
 	return services, nil
 }
 
+func (r *ReconcilePerconaServerMongoDB) exportServices(cr *api.PerconaServerMongoDB) error {
+	ls := clusterLabels(cr)
+
+	seList := mcs.ServiceExportList()
+	err := r.client.List(context.TODO(),
+		seList,
+		&client.ListOptions{
+			Namespace:     cr.Namespace,
+			LabelSelector: labels.SelectorFromSet(ls),
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, "get service export list")
+	}
+	if !cr.Spec.MultiCluster.Enabled {
+		for _, se := range seList.Items {
+			err = r.client.Delete(context.TODO(), &se)
+			if err != nil {
+				return errors.Wrap(err, "delete service export "+se.Name)
+			}
+		}
+		return nil
+	}
+
+	svcList := &corev1.ServiceList{}
+	err = r.client.List(context.TODO(),
+		svcList,
+		&client.ListOptions{
+			Namespace:     cr.Namespace,
+			LabelSelector: labels.SelectorFromSet(ls),
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, "get service list")
+	}
+
+	svcNames := make(map[string]struct{}, len(svcList.Items))
+	for _, svc := range svcList.Items {
+		se := mcs.ServiceExport(cr, svc.Name, ls)
+		err = setControllerReference(cr, se, r.scheme)
+		if err != nil {
+			return errors.Wrap(err, "set owner ref for serviceexport "+se.Name)
```

Comment:
please use `Wrapf` as below

Reply:
fixed in 7762104af1cb4ee68f3afa70caa7852fb0548d82